### PR TITLE
[BACKLOG-31357] Moved CE and EE big data drives to kar files and out …

### PR DESCRIPTION
**Pull request by angel-ramoscardona**
…of Karaf default deployment

### CODE REVIEW

**Comment by buildguy** [comment]
### :gear: Build finished in 7m 5s with command:
```
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd -Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/1/testReport)|
|----|

:information_source:  _This is an automatic message_

File: common-services-api/pom.xml [hunk 1]
```diff
@@ -12,10 +12,15 @@
     <packaging>bundle</packaging>
 
     <properties>
-      <vendor-driver.version>1.0.0.0</vendor-driver.version>
-      <pentaho-code.version>11.0.0.0</pentaho-code.version>
+      <org.osgi.core.version>4.3.1</org.osgi.core.version>
```
Comment (graimundo) [inline common-services-api/pom.xml]: Do not declare the osgi version here. Use the one from the parent pom: https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L381-L386

File: common-services-api/pom.xml [hunk 2]
```diff
@@ -12,10 +12,15 @@
...
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.core</artifactId>
+            <version>${org.osgi.core.version}</version>
```
Comment (graimundo) [inline common-services-api/pom.xml]: Do not explicitly use the version here. Use the managed dependency from the parent pom.

File: common-services-api/pom.xml [hunk 3]
```diff
@@ -62,14 +67,14 @@
...
-                        <Bundle-SymbolicName>pentaho-hadoop-driver-V1</Bundle-SymbolicName>
-                        <Bundle-Version>9.9.9.9</Bundle-Version>
-                        <Pentaho-Code-Version>${pentaho-code.version}</Pentaho-Code-Version>
                         <DynamicImport-Package>*</DynamicImport-Package>
                         <Import-Package>
                         </Import-Package>
                         <_exportcontents>
                         </_exportcontents>
+                        <Export-Package>
+                            org.pentaho.big.data.api.jdbc.impl
```
Comment (graimundo) [inline common-services-api/pom.xml]: If you want to export the package do not use the `impl` (or `internal`) package name as these are conventioned to be implementation details that should not be exported https://github.com/pentaho/pentaho-coding-standards/wiki/3.3.-OSGi-best-practices.

File: pom.xml [hunk 4]
```diff
@@ -51,6 +51,7 @@
     <htrace-core.version>3.1.0-incubating</htrace-core.version>
     <zookeeper.version>3.4.6.2.6.0.3-8</zookeeper.version>
     <dependency.maven-bundle-plugin.version>3.2.0</dependency.maven-bundle-plugin.version>
+    <karaf-maven-plugin.version>3.0.3</karaf-maven-plugin.version>
```
Comment (graimundo) [inline pom.xml]: Do not define a versio here. Use the karaf maven plugin from the plugin management in the parent pom.

File: shims/cdh514/kar/pom.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      <artifactId>pentaho-hadoop-shims-cdh514-driver</artifactId>
+      <version>9.0.0.0-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
```
Comment (graimundo) [inline shims/cdh514/kar/pom.xml]: Why not use the profile for features from the parent pom?
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1669
I believe that with that profile you should also be able to produce a kar as well without issues.

File: shims/cdh61/kar/pom.xml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      <artifactId>pentaho-hadoop-shims-cdh61-driver</artifactId>
+      <version>9.0.0.0-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
```
Comment (graimundo) [inline shims/cdh61/kar/pom.xml]: Why not use the profile for features from the parent pom?
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1669
I believe that with that profile you should also be able to produce a kar as well without issues.

File: shims/emr521/kar/pom.xml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      <artifactId>pentaho-hadoop-shims-emr521-driver</artifactId>
+      <version>9.0.0.0-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
```
Comment (graimundo) [inline shims/emr521/kar/pom.xml]: Why not use the profile for features from the parent pom?
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1669
I believe that with that profile you should also be able to produce a kar as well without issues.

File: shims/hdp26/kar/pom.xml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      <version>9.0.0.0-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
+        <extensions>true</extensions>
```
Comment (graimundo) [inline shims/hdp26/kar/pom.xml]: Why not use the profile for features from the parent pom?
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1669
I believe that with that profile you should also be able to produce a kar as well without issues.

File: shims/hdp30/kar/pom.xml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+      <artifactId>pentaho-hadoop-shims-hdp30-driver</artifactId>
+      <version>9.0.0.0-SNAPSHOT</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.karaf.tooling</groupId>
+        <artifactId>karaf-maven-plugin</artifactId>
+        <version>${karaf-maven-plugin.version}</version>
```
Comment (graimundo) [inline shims/hdp30/kar/pom.xml]: Why not use the profile for features from the parent pom?
https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L1669
I believe that with that profile you should also be able to produce a kar as well without issues.

File: common-fragment-V1/pom.xml [hunk 10]
```diff
@@ -548,6 +548,11 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.pentaho</groupId>
+            <artifactId>pentaho-hadoop-shims-common-services-api</artifactId>
```
Comment (graimundo) [inline common-fragment-V1/pom.xml]: Isn't this kind of introducing a circular dependency? Doesn't the `pentaho-hadoop-shims-common-services-api` depend on this module? 
The `pentaho-hadoop-shils-common-services-api` depends on `DriverLocatorImpl` that is declared in this module.
https://github.com/pentaho/pentaho-hadoop-shims/blob/82c0a8d19c2058dda50abdb3d19cd3f5cb1028c3/common-services-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L38
https://github.com/pentaho/pentaho-hadoop-shims/blob/master/common-fragment-V1/src/main/java/org/pentaho/big/data/api/jdbc/impl/DriverLocatorImpl.java

**Comment by buildguy** [comment]
### :gear: Build finished in 6m 49s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/3/testReport)|
|----|


:information_source:  _This is an automatic message_

File: common-services-api/pom.xml (hunk 1 above)
Comment (angel-ramoscardona) [reply common-services-api/pom.xml]: Replaced with parent pom's version

File: common-services-api/pom.xml (hunk 2 above)
Comment (angel-ramoscardona) [reply common-services-api/pom.xml]: Replaced with parent pom's version

File: common-services-api/pom.xml (hunk 3 above)
Comment (angel-ramoscardona) [reply common-services-api/pom.xml]: removed

File: common-fragment-V1/pom.xml (hunk 10 above)
Comment (angel-ramoscardona) [reply common-fragment-V1/pom.xml]: No. The dependency of pentaho-hadoop-shims-common-services-api on common-fragment-V1 is removed in these set of changes. Now only the common-fragment-V1 dependency on pentaho-hadoop-shims-common-services-api will remain.

**Comment by buildguy** [comment]
### :gear: Build finished in 6m 55s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/4/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 6m 34s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/5/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 6m 44s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/6/testReport)|
|----|


:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :gear: Build finished in 7m 1s with command:
```console
mvn clean verify -B -e -Daether.connector.resumeDownloads=false -amd \
-Daudit -P !assemblies
```
:ok_hand: All tests passed!
 ---

|Total|Failed|Skipped|
|---|---|---|
|858|0|0|

|[Details](http://wingman2.pentaho.net/job/projects/job/big-data-hadoop-shims/job/PR-1008/7/testReport)|
|----|


:information_source:  _This is an automatic message_

File: common-services-api/pom.xml (hunk 3 above)
Comment (graimundo) [reply common-services-api/pom.xml]: @pentaho/moseisley  @angel-ramoscardona you are still exporting classes from an `impl` package. You just substituted `impl` by `*`. 
https://github.com/pentaho/pentaho-hadoop-shims/tree/master/common-services-api/src/main/java/org/pentaho/big/data/api/jdbc/impl

If by design you want those classes to be exported, meaning that they are not implementation / internal details of the bundle as they are exposed to others, then they should be moved to another package. For example, moving them to the `org.pentaho.big.data.api.jdbc`package.

I didn't check the other `impl` packages. But if you are also exporting those somewhere, then they should also change their package.

**Pull request merged**